src/timedate/timedated.c

```diff
@@ -500,18 +707,26 @@ static int method_set_local_rtc(sd_bus_message *m, void *userdata, sd_bus_error
 }
 
 static int method_set_time(sd_bus_message *m, void *userdata, sd_bus_error *error) {
-        int relative, interactive;
+        sd_bus *bus = sd_bus_message_get_bus(m);
+        int relative, interactive, r;
         Context *c = userdata;
         int64_t utc;
         struct timespec ts;
         usec_t start;
         struct tm* tm;
-        int r;
 
         assert(m);
         assert(c);
 
-        if (c->use_ntp)
+        /* The list of NTP client service may not be perferct. So, at first check whether time is synced. */
+        if (ntp_synced())
```

Comment:
i wonder if this is safe to do actually... given that the flag is never turned off again if an ntp service had a sync once and then just died...